upstairs/src/volume.rs

```diff
@@ -1245,21 +1291,82 @@ mod test {
 
         let mut expected = vec![3; 512 * 2];
         expected.extend(vec![2; 512]);
-        expected.extend(vec![0; 512]); // <- was previously from parent
-        expected.extend(vec![0; 2048]);
+        expected.extend(vec![0; 512]); // <- was previously from parent, now "-"
+        expected.extend(vec![0x00; 512 * 4]); // <- from subvolume, still "-"
 
         assert_eq!(*buffer.as_vec(), expected);
 
+        // Write to the whole volume. There's no more read-only parent (it was
+        // dropped above)
+        volume
+            .write(
+                Block::new(0, block_size.trailing_zeros()),
+                Bytes::from(vec![9; 4096]),
+            )?
+            .block_wait()?;
+
+        let buffer = Buffer::new(4096);
+        volume
+            .read(Block::new(0, block_size.trailing_zeros()), buffer.clone())?
+            .block_wait()?;
+
+        assert_eq!(*buffer.as_vec(), vec![9; 4096]);
+
+        Ok(())
```

Comment:
I added a few that we talked about - did I miss one?

Reply:
No, hah.  No more tests are requested.

That was my late night half edited comment that makes no sense. 
I was wondering if the reason that the test you first posted in chat:
```
+        // Write to the whole volume
+        //
+        // parent blocks: 4 4 4 4
+        //    sub volume: 9 9 9 9 9 9 9 9
```
If that test did not reproduce the problem because it was added to the test function below the place where the subvolume was removed?

Reply:
I think you're right, there's no way it would test it there.